tests/unit/amo/sagas/test_addonsByAuthors.js

```diff
@@ -41,14 +43,85 @@ describe(__filename, () => {
   function _fetchAddonsByAuthors(params) {
     sagaTester.dispatch(fetchAddonsByAuthors({
       errorHandlerId: errorHandler.id,
-      addonType: ADDON_TYPE_THEME,
       ...params,
     }));
   }
 
   it('calls the API to retrieve other add-ons', async () => {
+    const fakeAddons = [{ ...fakeAddon, id: 54, slug: 'fifty-four' }];
+    const fakeThemes = [
+      { ...fakeAddon, id: 55, slug: 'fifty-five', type: ADDON_TYPE_THEME },
+    ];
+    const authorNames = ['mozilla', 'johnedoe'];
+    const state = sagaTester.getState();
+
+    mockApi
+      .expects('search')
+      .withArgs({
+        api: state.api,
+        filters: {
+          addonType: ADDON_TYPE_THEME,
+          author: authorNames.sort().join(','),
+          exclude_addons: undefined, // `callApi` will internally unset this
+          page_size: ADDONS_BY_AUTHORS_PAGE_SIZE,
+          sort: SEARCH_SORT_TRENDING,
+        },
+      })
+      .once()
+      .returns(Promise.resolve(createAddonsApiResult(fakeThemes)));
+
+    mockApi
+      .expects('search')
+      .withArgs({
+        api: state.api,
+        filters: {
+          addonType: ADDON_TYPE_EXTENSION,
+          author: authorNames.sort().join(','),
+          exclude_addons: undefined, // `callApi` will internally unset this
+          page_size: ADDONS_BY_AUTHORS_PAGE_SIZE,
+          sort: SEARCH_SORT_TRENDING,
+        },
+      })
+      .once()
+      .returns(Promise.resolve(createAddonsApiResult(fakeAddons)));
+
+    _fetchAddonsByAuthors({ addonType: ADDON_TYPE_THEME, authorNames });
+    _fetchAddonsByAuthors({ addonType: ADDON_TYPE_EXTENSION, authorNames });
+
+    const loadActionForAddons = loadAddonsByAuthors({
+      addons: fakeAddons,
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames,
+      forAddonSlug: undefined,
+    });
+    const loadActionForThemes = loadAddonsByAuthors({
+      addons: fakeThemes,
+      addonType: ADDON_TYPE_THEME,
+      authorNames,
+      forAddonSlug: undefined,
+    });
+
+    await sagaTester.waitFor(loadActionForAddons.type);
+    mockApi.verify();
+
+    let matchingActions = 0;
+    sagaTester.getCalledActions().filter((action) => {
+      return action.type === loadActionForAddons.type;
```

Comment:
I am confused by this. Won't this `filter` only allow actions with a `loadActionForAddons` type, but we are also looking for actions with a `loadActionForThemes` type? Or do both of those have the same `type`? I wonder if there's a less confusing way to write this, as it hurts my head. :(